### ISSUE TITLE
Unable to reproduce adversarial training results

### ISSUE THREAD

**Issue by ClonedOne** [Trusted-AI/adversarial-robustness-toolbox#907]
**Describe the bug**
I am using `AdversarialTrainer`, as mentioned in [this example notebook](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/82727cb56975a1a41eca86e1ef669ad675646cc7/notebooks/adversarial_training_mnist.ipynb), with a PGD attack (code below) to train a simple CNN. Despite not throwing any error during training, the model appears to be equivalent to an untrained model -- extremely low accuracy both on normal test data and under attack.

Dataset: MNIST

I have also tried with AdversarialTrainerMadryPGD with default parameters, to check if there was anything wrong with the attack parameters I was using, but I get the same results.

I am probably doing something wrong, but from the example notebooks and scripts I couldn't find any indication of what I should have done differently.

**To Reproduce**

```python
model = build_model(input_shape=(28, 28, 1), num_classes=10)

# Wrappers for ART
classifier = KerasClassifier(
    model=model,
    clip_values=(min_, max_),
    use_logits=False
)
# Attacker object
pgd = ProjectedGradientDescent(
    classifier,
    eps=0.3,
    eps_step=0.01,
    max_iter=40
)

# Create adversarial trainer and perform adversarial training
trainer = AdversarialTrainer(
    classifier,
    pgd,
    ratio=1.0
)

trainer.fit(x_train_s, y_train_s, nb_epochs=n_epochs, batch_size=batch_size)
```

**Expected behavior**
The model is trained correctly, and is also adversarially robust --> high accuract on test set, non-trivial accuracy under PGD attack.

**Screenshots**
These are the full notebooks:
 - https://github.com/ClonedOne/exp_nbs/blob/master/advtrain_pgd.ipynb
 - https://github.com/ClonedOne/exp_nbs/blob/master/advtrain_madry.ipynb

**System information (please complete the following information):**
 - OS: Ubuntu 18.04
 - Python version: 3.8.5
 - ART version or commit number: 1.5.1
 - TensorFlow / Keras / PyTorch / MXNet version: 2.3.0 / 2.4.3 / _ / _

Thank you!

**Comment by beat-buesser** [comment]
Hi @ClonedOne Thank you very much for raising this issue and providing a detailed notebook! I'll take a closer look.

**Comment by beat-buesser** [comment]
Hi @ClonedOne

Your first [notebook](https://github.com/ClonedOne/exp_nbs/blob/master/advtrain_pgd.ipynb) is already very close to the experiments of Madry et al. (https://arxiv.org/abs/1706.06083) for MNIST but with 2 small but critical differences:
- they use PGD with random initialisation (`num_random_init=1`) instead of PGD without random initialisation
- they use a learning rate of 0.0001 (`optimizer=keras.optimizers.Adam(learning_rate=0.0001)`) instead of 0.001

Your second [notebook](https://github.com/ClonedOne/exp_nbs/blob/master/advtrain_madry.ipynb) uses `AdversarialTrainerMadryPGD` with its default values which are according to the docstring for CIFAR-10 in pixel range 0-255 whereas your notebook uses MNIST in pixel range 0-1.

**Comment by ClonedOne** [comment]
Hello @beat-buesser, thank you for your reply! I am going to experiment with the parameters you are suggesting. From a couple of initial tests, it seems that lowering the learning rate is the key to make the training succeed.

**Comment by beat-buesser** [comment]
Hi @ClonedOne Great, please let me know what you find out in your experiments!

**Comment by ClonedOne** [comment]
Thank you again for the clarification, these parameters indeed make the training converge, and be adversarially robuts. The key factor here was, by far, the lower learning rate. I would suggest, if possible, to add a note on this in the documentation.

Thanks!
